### PR TITLE
Add sidebar component

### PR DESCRIPTION
**Pull request by Ajotah98**
I updated the component "Sidebar" started on March.

### CODE REVIEW

File: sqlpage/templates/shell.handlebars [hunk 1]
```diff
@@ -1,178 +1,267 @@
...
+                    aria-controls="navbar-menu" aria-expanded="false" aria-label="Toggle navigation">
+                    <span class="navbar-toggler-icon"></span>
+                </button>
+                <div class="collapse navbar-collapse" id="navbar-menu">
+                    <ul class="navbar-nav ms-auto">
+                        {{#each (to_array menu_item)}}
+                        {{#if (or (eq (typeof this) 'object') (and (eq (typeof this) 'string') (starts_with this
+                        '{')))}}
+                        {{#with (parse_json this)}}
+                        {{#if (or (or this.title this.icon) this.image)}}
```
Comment (lovasoa) [inline sqlpage/templates/shell.handlebars:177]: I'm a little afraid of duplicating this logic... Do you think we could use the same markup for the horizontal and the vertical sidebars, with only classes differing between the two ? Or would that be impossible ?

**Review by lovasoa — COMMENTED** [review]
Cool, thank you very much for the PR ! I want to get it through the finish line ! Can you have a look at my comment above, and revert the indentation changes  ?

**Comment by Ajotah98** [comment]
> Cool, thank you very much for the PR ! I want to get it through the finish line ! Can you have a look at my comment above, and revert the indentation changes ?

Sorry! I was busy. I'll update it asap and review the comment. Thanks!

**Comment by Ajotah98** [comment]
Hi! I was seeing the changes you have done and I really think... That is all done for now? I don't know if you want to implement more things onto the sidebar

**Comment by lovasoa** [comment]
As I said, I'm a little bit hesitant of duplicating the logic of parsing and iterating over the menu items. Is there any chance you can factor it in order to have the same markup for the horizontal and vertical bars, with only classes differing?

**Review by lovasoa — APPROVED** [review]
Great, thank you very much !

**Comment by lovasoa** [comment]
I think this is ready :) Can you please resolve the conflicts so I can merge ?

**Comment by lovasoa** [comment]
The conflicts probably come from the changes made here to fix the elision of long titles:

https://github.com/lovasoa/SQLpage/commit/6cc4cce73c1ac67dac3d18b932ab30983a11b933#diff-255998f398cb0bb6096065f84876aea7a2ca35ba2c6dda8162e4195e2654ee94

**Comment by Ajotah98** [comment]
Okey, I was seeking whats causing merge conflicts, and testing the actual menu-items from the commit you posted, and:

image

With this new approach, the sidebar menu items get aligned into the right as you can see in the image.

I'll check how can I make if it's on sidebar, apply another CSS styles to make the layout more flex

**Comment by Ajotah98** [comment]
I'm more convinced that the conflicts are now solved. The conflicts showed on the branch are a white-space that no longer exists on the component and the addition of the inline of menu-items

**Comment by lovasoa** [comment]
You can merge the main branch of this repo into your branch, then, and the conflicts will disappear here

**Comment by Ajotah98** [comment]
Cool, it worked. Thanks!

**Comment by lovasoa** [comment]
Great, here we are! Thank you!

**Comment by lovasoa** [comment]
Argh, I think this broke the menu layout on narrow screens :/

before

image


after

image

**Comment by lovasoa** [comment]
Also there is now an underline on the page title on hover

image

**Comment by Ajotah98** [comment]
I'm going to check the CSS. Sorry for the inconvenience

**Comment by Ajotah98** [comment]
> Also there is now an underline on the page title on hover
> 
> image

This behaviour occurs in v.0.26.0. Is not a modification on the sidebar addition :/

**Comment by lovasoa** [comment]
Thank you ! No problem. You can have a look at the mobile layout problem, and I'll take care of the underline, then !

**Comment by Ajotah98** [comment]
I found the problem! I'll try to fix it ASAP

**Comment by lovasoa** [comment]
I'm updating the https://sql.ophir.dev/examples/layouts.sql page to demonstrate the new sidebar

**Pull request merged**